salt/states/xml.py

```diff
@@ -54,20 +67,216 @@ def value_present(name, xpath, value, **kwargs):
     current_value = __salt__["xml.get_value"](name, xpath)
     if not current_value:
         ret["result"] = False
-        ret["comment"] = "xpath query {0} not found in {1}".format(xpath, name)
+        ret["comment"] = "xpath query {} not found in {}".format(xpath, name)
         return ret
 
     if current_value != value:
         if kwargs["test"]:
             ret["result"] = None
-            ret["comment"] = "{0} will be updated".format(name)
+            ret["comment"] = "{} will be updated".format(name)
             ret["changes"] = {name: {"old": current_value, "new": value}}
         else:
             results = __salt__["xml.set_value"](name, xpath, value)
             ret["result"] = results
-            ret["comment"] = "{0} updated".format(name)
+            ret["comment"] = "{} updated".format(name)
             ret["changes"] = {name: {"old": current_value, "new": value}}
     else:
-        ret["comment"] = "{0} is already present".format(value)
+        ret["comment"] = "{} is already present".format(value)
+
+    return ret
+
+
+def contains_tree(name, fragment, xpath=None):
+    """
+    .. versionadded:: NEXT_RELEASE
+
+    Merge a block of XML into another, in a given file. This is used for
+    ensuring a tree exists, or tweaking attributes of a well-known location in
+    a config file. This state only matches tags as it merges fragments, so it
+    does not work well (if at all) with list-style data.
+    This will NOT merge text/tail nodes.
+
+    name
+        The location of the XML file to manage, as an absolute path.
+
+    xpath
+        optional xpath location under which to place the fragment. See
+        https://docs.python.org/3/library/xml.etree.elementtree.html#example
+        to see what syntax is supported.
+
+    fragment
+        XML fragment to create _under_ the location specified in `path`.
+        The fragment must be valid XML. If the path already contains XML and
+        the fragment's tags match, they will be merged -- the fragment's
+        properties will overwrite any existing properties with the same key.
+        This means attributes are merged, text, and tail values are replaced.
+
+    .. code-block:: yaml
+
+        ensure_tree_exists:
+        xml.contains_tree:
+            - name: /etc/data.xml # MUST already have valid XML.
+            - resolve: merge
```

Comment:
Looks good!

The resolve key appears to no longer be used, so it can be removed from the docs.

Reply:
Fixed! Thanks for noticing.